core/src/main/scala/caliban/schema/Schema.scala

```diff
@@ -313,7 +315,8 @@ trait DerivationSchema[R] {
 
   def combine[T](ctx: ReadOnlyCaseClass[Typeclass, T]): Typeclass[T] = new Typeclass[T] {
     override def toType(isInput: Boolean = false): __Type =
-      if (isInput)
+      if (ctx.isValueClass) ctx.parameters.headOption.map(_.typeclass.toType(isInput)).get
```

Comment:
We can make `ctx.parameters.head.typeclass.toType(isInput)` instead. I guess that would be more efficient but unsafe. The las `.get` is unsafe also, but a little bit less unsafe.

Reply:
It shouldn't be possible to have anything else than 1 parameter but just in case, maybe you can add a check in the `if`? Then you can just use `head` safely.